### ISSUE TITLE
 mbox layer has some question

### ISSUE THREAD

**Issue by Soleil-hao** [weiliu89/caffe#587]
Please use the [caffe-users list](https://groups.google.com/forum/#!forum/caffe-users) for usage, installation, or modeling questions, or other requests for help.
_Do not post such requests to Issues._ Doing so interferes with the development of Caffe.

Please read the [guidelines for contributing](https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md) before submitting this issue.

### Issue summary
When i train the caffe model,  at the time of iteration 1000, testing net get the core dumped:
solver.cpp:464] Check failed: result[j]->width() == 5 (150 vs. 5) 
at  the SSD Solver.hpp, 
Dtype iter_loss;
    const vector<Blob<Dtype>*>& result = test_net->Forward(&iter_loss);
    if (param_.test_compute_loss()) {
      loss += iter_loss;
    }
    for (int j = 0; j < result.size(); ++j) {
      CHECK_EQ(result[j]->width(), 5);
      const Dtype* result_vec = result[j]->cpu_data();
      int num_det = result[j]->height();
      for (int k = 0; k < num_det; ++k) {
        int item_id = static_cast<int>(result_vec[k * 5]);
        int label = static_cast<int>(result_vec[k * 5 + 1]);

i want to know the meaning of the result[j]->width(), and why check with 5.
thanks a lot!
 
### Steps to reproduce

If you are having difficulty building Caffe or training a model, please ask the caffe-users mailing list. If you are reporting a build error that seems to be due to a bug in Caffe, please attach your build configuration (either Makefile.config or CMakeCache.txt) and the output of the make (or cmake) command.

### Your system configuration
Operating system:
Compiler:
CUDA version (if applicable):
CUDNN version (if applicable):
BLAS:
Python or MATLAB version (for pycaffe and matcaffe respectively):

**Comment by weiliu89** [comment]
You have some other output except the `detection_out`. Use [this tool](http://ethereon.github.io/netscope/#/editor) to check.

**Comment by liangxi627** [comment]
Hello, @Soleil-hao , have you solved the problem? I have this error when I add some extra convolutional layers, I want to know how to check the result[j], since I didn't define new output except the detection_out. @weiliu89 Please help me.
>  Check failed: result[j]->width() == 5 (3 vs. 5)

**Comment by shiva13425** [comment]
@liangxi627 Am facing 
**Check failed: result[j]->width() == 5 (3 vs. 5)** error. 
Kindly share your comments to fix it
Thanks

**Comment by liangxi627** [comment]
I just used the tool provided by weiliu89 to check the network structure, and indeed have two conv layers as output except the detection_out.
